Content.Server/Anomaly/Effects/ChasingAnomalySystem.cs

```diff
@@ -0,0 +1,79 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Anomaly.Components;
+using Content.Shared.Anomaly.Components;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Random;
+
+namespace Content.Server.Anomaly.Effects;
+/// <summary>
+/// This component allows the anomaly to chase a random instance of the selected type component within a radius.
+/// </summary>
+public sealed class ChasingAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ChasingAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+    }
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var (anom, trans) in EntityManager.EntityQuery<ChasingAnomalyComponent, TransformComponent>(true))
+        {
+            if (!Initialized(anom.Owner)) return;
+            if (Deleted(anom.ChasingEntity)) continue;
+            if (!anom.ChasingEntity.IsValid()) continue;
+            if (anom.ChasingEntity == default!) continue;
+
+            //Calculating direction to the target.
+            var xformQuery = GetEntityQuery<TransformComponent>();
```

Comment:
make this a system field and set it in the initialize function

`private EntityQuery<TransformComponent> _xformQuery;`
`_xformQuery = GetEntityQuery<TransformComponent>();`